### PR TITLE
webapp: Set-AzureRmWebApp doesn't handle existing Tags when updating web site

### LINKED ISSUE #8884: Set-AzureRmWebApp doesn't handle existing Tags when updating web site
## Description

The existing tags will be remove after running `Set-AzureRmWebApp` 

https://github.com/Azure/azure-powershell/blob/3ef406acd59be744bcc345631b10f464a1d53bf9/src/Websites/Websites/Utilities/WebsitesClient.cs#L99

`CreateOrUpdate` will `PUT` the new web app object which will remove the exisitng tags.

## Steps to reproduce

```powershell
Set-AzureRmWebApp -Name $name -ResourceGroupName $resourcegroup -AssignIdentity $true
```

## Module versions

```powershell

```

## Debug output

```

```

## Error output

```

```

### PR DESCRIPTION
**Pull request by panchagnula**
Fixes #8884 

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
@panchagnula whenever you're able to make the above change, would you also pull the latest from the `master` branch? We made a change in that branch that needs to go into any PR currently targeting it.

**Comment by panchagnula** [comment]
> @panchagnula whenever you're able to make the above change, would you also pull the latest from the `master` branch? We made a change in that branch that needs to go into any PR currently targeting it.

@cormacpayne , i rebased to master -> & tried updating the test by adding this line to add tags 	Set-AzResource -ResourceType Microsoft.Web/sites -ResourceGroupName $rgname -ResourceName $webAppName -Tag @{key0="value0"} , however when i try to run the test for recording i see the error, that this command is not found, do i need to import this module to the appservice project? Any examples here will be helpful. Thanks!

**Comment by cormacpayne** [comment]
@panchagnula I believe we removed the script that some test projects were using to call select Resources commands, so you'll need to add a reference to both of the Resources projects ([`Resources.csproj`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/Resources.csproj) and [`ResourceManager.csproj`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/ResourceManager/ResourceManager.csproj)) to your solution so the `Az.Resources.psd1` module will get built when your solution is built. Then you'll need to make sure the `Az.Resources.psd1` file is being imported in your test controller. Feel free to ping me if you have any questions.

**Comment by panchagnula** [comment]
@cormacpayne thanks for the pointers. On the websites project i see a reference to this https://github.com/Azure/azure-powershell/blob/master/tools/ScenarioTest.ResourceManager/AzureRM.Resources.ps1

However, the file doesn't have a function definition for Set-AzureRmResource or Set-AzResource , could this be it. Is there any guidelines i need to follow to update this!

**Comment by cormacpayne** [comment]
@panchagnula that's the file we use for a few generic Resources calls in cmdlets. If you still plan on using cmdlets from `Az.Resources` in your tests, I would still advise to reference the two projects in your solution and import the `Az.Resources.psd1` in your test controller.

File: src/Websites/Websites/Utilities/WebsitesClient.cs [hunk 1]
```diff
@@ -816,11 +817,10 @@ public Site UpdateHostNameSslState(string resourceGroupName, string webAppName,
                     ToUpdate = true,
                     SslState = sslState
                 }},
-                Location = location
+                Location = location,
+                Tags =  webapp.Tags
```
Comment (cormacpayne) [inline src/Websites/Websites/Utilities/WebsitesClient.cs]: @panchagnula we should preserve the null-check here that we had previously (`webapp?.Tags`)

Comment (panchagnula) [reply src/Websites/Websites/Utilities/WebsitesClient.cs]: i will add that back.

**Comment by panchagnula** [comment]
@cormacpayne , i added the tests for tags, however changing to use the new Resources & ResourceManager - i cannot run any tests because of a failure in EnvironmentSetupHelper execution. If this something we need to do i will handle this in a separate PR but wanted to unblock this PR.

**Comment by panchagnula** [comment]
@maddieclayton or @cormacpayne  can one of you point to me what revision needs to be made here please? 
I see the BUILD fails but i am not sure why - thank you.

**Comment by maddieclayton** [comment]
@panchagnula Can you try pulling from upstream and pushing the merge commit? It looks like a SQL test is failing which is not related to your change.

**Comment by panchagnula** [comment]
> @panchagnula Can you try pulling from upstream and pushing the merge commit? It looks like a SQL test is failing which is not related to your change.

@maddieclayton , merge to upstream master did not seem to fix it.

**Comment by maddieclayton** [comment]
Failure now is a websites test:
Failed   Microsoft.Azure.Commands.Websites.Test.ScenarioTests.SSLBindingTests.TestRemoveNewWebAppSSLBinding
EXEC : error Message:  [D:\a\1\s\build.proj]
   System.Management.Automation.CmdletInvocationException : Unable to find a matching HTTP request for URL 'GET /subscriptions/798436ab-a054-4fbf-ba01-f9f8cce4e65f/resourceGroups/lketmtestantps10/providers/Microsoft.Web/sites/lketmtestantps10?api-version=2018-02-01'. Calling method Item().
  ---- System.Collections.Generic.KeyNotFoundException : Unable to find a matching HTTP request for URL 'GET /subscriptions/798436ab-a054-4fbf-ba01-f9f8cce4e65f/resourceGroups/lketmtestantps10/providers/Microsoft.Web/sites/lketmtestantps10?api-version=2018-02-01'. Calling method Item().
  Stack Trace:

This message usually means a change has been made to a test without rerecording, please try rerecording that test.

**Pull request merged**